src/main/server.ts

```diff
@@ -29,9 +29,57 @@ import {
 import log from 'electron-log';
 
 import * as fs from 'fs';
+import * as os from 'os';
+import * as  path from 'path';
 import { IPythonEnvironment } from './tokens';
 import { appConfig } from './utils';
 
+function createTempFile(fileName = 'temp', data = '', encoding: BufferEncoding = 'utf8') {
+    const tempDirPath = path.join(os.tmpdir(), 'jlab_desktop');
+    const tmpDir = fs.mkdtempSync(tempDirPath);
+    const tmpFilePath = path.join(tmpDir, fileName);
+
+    fs.writeFileSync(tmpFilePath, data, {encoding});
+
+    return tmpFilePath;
+}
+
+function createLaunchScript(environment: IPythonEnvironment): string {
+    const platform = process.platform;
+    const isWin = platform === 'win32';
+    const pythonPath = environment.path;
+    let envPath = path.dirname(pythonPath);
+    if (!isWin) {
+        envPath = path.normalize(path.join(envPath, '../'));
+    }
+
+    // note: traitlets<5.0 require fully specified arguments to
+    // be followed by equals sign without a space; this can be
+    // removed once jupyter_server requires traitlets>5.0
+    const launchCmd = `python -m jupyterlab --no-browser --JupyterApp.config_file_name="" --ServerApp.port=${appConfig.jlabPort} --ServerApp.password="" --ServerApp.allow_origin="*" --ContentsManager.allow_hidden=True`;
```

Comment:
One neat way to preserve the comments on why we need particular options (which will otherwise become forgotten) 
would be to have the arguments as a list of strings and concatenate once ready.